File: app/assets/javascripts/boards/models/milestone.js
export default class ListMilestone {
  constructor(obj) {
    this.id = obj.id;
    this.title = obj.title;
    this.path = obj.path;
    this.state = obj.state;
    this.webUrl = obj.web_url || obj.webUrl;
    this.description = obj.description;
  }
}

window.ListMilestone = ListMilestone;

File: lib/api/entities/milestone.rb
# frozen_string_literal: true

module API
  module Entities
    class Milestone < Grape::Entity
      expose :id, :iid, documentation: { type: 'Integer' }
      expose :project_id, documentation: { type: 'Integer' }, if: ->(entity, options) { entity&.project_id }
      expose :group_id, if: ->(entity, options) { entity&.group_id }
      expose :title, :description
      expose :state, :created_at, :updated_at
      expose :due_date
      expose :start_date
      expose :expired, documentation: { type: 'Boolean' }

      expose :web_url do |milestone, _options|
        Gitlab::UrlBuilder.build(milestone)
      end
    end
  end
end
